### PR TITLE
api practice homework with 100+ users and products

### PR DESCRIPTION
**Pull request by jpmarcantonio**

### CODE REVIEW

File: Create_users_on_website.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import argparse
+# import sys
+# arguments = sys.argv
+# number_of_users = arguments[1]
+parser = argparse.ArgumentParser(description='Process some integers.')
+parser.add_argument('--number_of_users', required=True,
+                    help='how many users to create')
+args = parser.parse_args()
+num_of_users = args.number_of_users
+
```
Comment (admaskinfu) [inline Create_users_on_website.py:21]: You should get the credentials from environment variables.

File: Create_users_on_website.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+wcapi = API(
+    url="http://localhost:8888/mysite2",
+    consumer_key="<add consumer key>",
+    consumer_secret="<add consumer secret>",
+    version="wc/v3",
+    timeout = 60
+)
+rand_pass = str(uuid.uuid4())
+for i in range(int(num_of_users)):
```
Comment (admaskinfu) [inline Create_users_on_website.py:55]: Can you please log number of users created everytime a new one is creted.

File: Frontiers_186_list_of_customer_emails.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    customers = r.json()
+    output_file = 'list_of_user_email.csv'
+
+    page += 1
+    if not customers:
+        break
+
+    for customer in customers:
+        customer_email = customer["email"]
+        email_list.append([customer_email])
```
Comment (admaskinfu) [inline Frontiers_186_list_of_customer_emails.py]: Why are you adding alist to the list of email. Why not add the email. Now you have a list of lists, not a list of strings.
I would' ven done email_list.append(customer_email)
But I think you writer.writerows() might change too if you make this change.

File: Frontiers_188_create_list_of_products_and_prices.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+product_list = []
+page = 1
+
+while True:
+    r = wcapi.get("products", params={"per_page": 100, 'page': page})
+    products = r.json()
+
+    status_code = r.status_code
+    if status_code != 200:
+        raise Exception(f"Expected status code 200. Got {status_code}")
```
Comment (admaskinfu) [inline Frontiers_188_create_list_of_products_and_prices.py]: In your error message you should print the response body too. the status code will tell you there is an issue but the body will tell you what the issue is. So in the error message add 'r.content'.
You can add r.json() too but .content always prints info .json() assumes the response is json.

File: create_products_on_website.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        "name": random_string,
+        "regular_price": str(random_float),
+        "price": random_float
+    }
+
+    rs_api = wcapi.post("products", data)
+    rs_json = rs_api.json()
+    name = rs_json['name']
+    price = rs_json['price']
+    print(name)
```
Comment (admaskinfu) [inline create_products_on_website.py]: perfect code. juat make the print statements meaningful.

File: Create_users_on_website.py (hunk 1 above)
Comment (jpmarcantonio) [reply Create_users_on_website.py:21]: Set these up and it works.

File: Create_users_on_website.py (hunk 2 above)
Comment (jpmarcantonio) [reply Create_users_on_website.py:55]: Added a counter to the script that outputs to the .log file and prints to the display.

File: Create_users_on_website.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                    help='how many users to create')
+args = parser.parse_args()
+num_of_users = args.number_of_users
+
+logging.basicConfig(filename='create_users.log', level=logging.DEBUG,
+                    format='%(asctime)s - %(message)s', datefmt='%d-%b-%y %H:%M:%S')
+
+api_key = os.getenv("MYSITE2_API_KEY")
+api_secret = os.getenv("MYSITE2_API_SECRET")
+
```
Comment (admaskinfu) [inline Create_users_on_website.py:46]: Add a code here that will check if the variables are set and if not raise an exception telling the user the variable name must be set. 
Other wise when the API fails it is not so easy to know why it failed.

File: Create_users_on_website.py (hunk 2 above)
Comment (admaskinfu) [reply Create_users_on_website.py:55]: The counter works fine. You did great with that.
But could you just use 'i' for the counter? Isn't "i" just a number?

Comment (jpmarcantonio) [reply Create_users_on_website.py:55]: replaced the counter with i. makes it  a lot cleaner and fewer lines of code. Is this the same i thats being used in the for loop?

File: Create_users_on_website.py (hunk 6 above)
Comment (jpmarcantonio) [reply Create_users_on_website.py:46]: created a function to do this. Using try and except to raise the exception. Tried to get the exception to read well in the logs and the console. Created new variables `variable_key and variable_secret` so the variable names only have to be hard coded once.

File: Frontiers_186_list_of_customer_emails.py (hunk 3 above)
Comment (jpmarcantonio) [reply Frontiers_186_list_of_customer_emails.py]: So I used chat gpt to help me with this. As it is if I change it to email_list.append(customer_email) the writer.writerows() will write each character separated by a comma.  to get around this I used a for loop to split each item at the comma (thanks again chat gpt)

File: Create_users_on_website.py (hunk 2 above)
Comment (jpmarcantonio) [reply Create_users_on_website.py:55]: answering my own question: it is definitely the same i !

File: Frontiers_188_create_list_of_products_and_prices.py (hunk 4 above)
Comment (jpmarcantonio) [reply Frontiers_188_create_list_of_products_and_prices.py]: This is handy, thanks. Is this what you had in mind?

File: create_products_on_website.py (hunk 5 above)
Comment (jpmarcantonio) [reply create_products_on_website.py]: Thanks man. Updated the print statement.

**Pull request merged**